hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/ShowBootstrapPartitionsProcedure.scala

```diff
@@ -42,14 +43,19 @@ class ShowBootstrapPartitionsProcedure extends BaseProcedure with ProcedureBuild
     super.checkArgs(PARAMETERS, args)
 
     val tableName = getArgValueOrDefault(args, PARAMETERS(0))
+    val limit = getArgValueOrDefault(args, PARAMETERS(1))
 
     val basePath: String = getBasePath(tableName)
     val metaClient = HoodieTableMetaClient.builder.setConf(jsc.hadoopConfiguration()).setBasePath(basePath).build
 
     val indexReader = createBootstrapIndexReader(metaClient)
     val indexedPartitions = indexReader.getIndexedPartitionPaths
 
-    indexedPartitions.stream().toArray.map(r => Row(r)).toList
+    if (limit.isDefined) {
+      indexedPartitions.stream().limit(limit.get.asInstanceOf[Int]).toArray.map(r => Row(r)).toList
+    } else {
```

Comment:
There could many partitions here? It is still makes sense to represent pruned result list?

Reply:
make sense，I had revert the method，ptal @danny0405